lib/archive.ml

```diff
@@ -111,32 +130,33 @@ let tar dir ~exclude_paths ~root ~mtime =
     | Some file -> Fpath.(root // file)
     in
     Logs.info (fun m -> m "Archiving %a" Fpath.pp fname);
-    tar
-    >>= fun tar -> OS.Dir.exists file
-    >>= function
+    tar >>= fun tar ->
+    OS.Dir.exists file >>= function
     | true -> Tar.add tar fname ~mode:0o775 ~mtime `Dir
     | false ->
-        OS.Path.Mode.get file
-        >>= fun mode -> OS.File.read file
-        >>= fun contents ->
+        OS.Path.Mode.get file >>= fun mode ->
+        OS.File.read file >>= fun contents ->
         let mode = if 0o100 land mode > 0 then 0o775 else 0o664 in
         Tar.add tar fname ~mode ~mtime (`File contents)
   in
-  path_set_of_dir dir ~exclude_paths
-  >>= fun fset -> Fpath.Set.fold tar_add fset (Ok Tar.empty)
-  >>| fun tar -> Tar.to_string tar
+  path_set_of_dir dir ~exclude_paths >>= fun fset ->
+  Fpath.Set.fold tar_add fset (Ok Tar.empty) >>| fun tar ->
+  Tar.to_string tar
 
 (* Bzip2 compression and unarchiving *)
 
 let bzip2_cmd = OS.Env.(value "DUNE_RELEASE_BZIP2" cmd ~absent:(Cmd.v "bzip2"))
+
 let ensure_bzip2 () = OS.Cmd.must_exist bzip2_cmd >>| fun _ -> ()
 
 let bzip2 ~dry_run ?force ~dst s =
   Sos.run_io ~dry_run ?force ~default:()
     bzip2_cmd (OS.Cmd.in_string s) (OS.Cmd.to_file dst)
 
 let tar_cmd = OS.Env.(value "DUNE_RELEASE_TAR" cmd ~absent:(Cmd.v "tar"))
-let ensure_tar () = OS.Cmd.must_exist tar_cmd >>| fun _ -> ()
+
+let ensure_tar () = OS.Cmd.must_exist tar_cmd >>| fun (_ : Cmd.t) -> ()
```

Comment:
Why is this annotation needed?

Reply:
(reverted)
it's just an habit I got after working on ocamlformat, it can be moved to a cleaning PR later